rotkehlchen/tests/unit/test_trades.py

```diff
@@ -148,3 +149,45 @@ def test_query_trade_history_online_but_exchange_excluded(events_historian):
 
     assert latest_trades_mock.call_count == 0
     assert iterate_exchanges_mock.call_count == 0
+
+
+@pytest.mark.parametrize('db_settings', [
+    {'non_syncing_exchanges': [ExchangeLocationID(name='Binance', location=Location.BINANCE)]}])
+def test_query_trade_history_with_exchange_instance_excluded(events_historian):
+    """Test that when ignoring the locations along with names, only the ignored location with that
+    name should be ignored and not all the instances of that location."""
+
+    def get_binances(name: list[str]) -> list[ExchangeInterface]:
+        return [
+            ExchangeInterface(
+                name,
+                Location.BINANCE,
+                ApiKey(''),
+                ApiSecret(b''),
+                events_historian.db,
+            )
+            for name in name
+        ]
+
+    binances = get_binances(['Binance', 'Binance2'])
+    patch_binance = patch.object(binances[0], 'query_trade_history')
+    patch_binance2 = patch.object(binances[1], 'query_trade_history')
+    mock_exchanges = {Location.BINANCE: binances}
+    patch_connected_exchange = patch.object(
+        events_historian.exchange_manager,
+        'connected_exchanges',
+        mock_exchanges,
+    )
```

Comment:
Don't create variables that you don't need

```suggestion
    patch_connected_exchange = patch.object(
        events_historian.exchange_manager,
        'connected_exchanges',
        {Location.BINANCE: binances},
    )
```

also can you use names for the arguments here?